File: src/main/java/com/dh/ondot/schedule/infra/exception/OdsayUnhandledException.java
package com.dh.ondot.schedule.infra.exception;

import com.dh.ondot.core.exception.InternalServerException;

import static com.dh.ondot.core.exception.ErrorCode.ODSAY_UNHANDLED_ERROR;

public class OdsayUnhandledException extends InternalServerException {
    public OdsayUnhandledException(String detail) {
        super(ODSAY_UNHANDLED_ERROR.getMessage().formatted(detail));
    }
    @Override
    public String getErrorCode() {
        return ODSAY_UNHANDLED_ERROR.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/exception/OdsayUnhandledException.kt
package com.dh.ondot.schedule.infra.exception

import com.dh.ondot.core.exception.ErrorCode.ODSAY_UNHANDLED_ERROR
import com.dh.ondot.core.exception.InternalServerException

class OdsayUnhandledException(detail: String) :
    InternalServerException(ODSAY_UNHANDLED_ERROR.message.format(detail)) {
    override val errorCode: String get() = ODSAY_UNHANDLED_ERROR.name
}
